fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/mixin/dimension/LevelStorageMixin.java

```diff
@@ -0,0 +1,109 @@
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.InvalidIdentifierException;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryLookupCodec;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.gen.GeneratorOptions;
+import net.minecraft.world.level.storage.LevelStorage;
+
+/*
+ * This is a bug fix.
+ * This fixes an issue where removing a mod or datapack that adds a dimension will cause every dimension but the overworld to also be disposed of.
+ *
+ * Testing:
+ * A good way to test this is to create a world with the dimension test mod enabled and then without to simulate a dimension removal.
+ *
+ * Tracking issue: https://bugs.mojang.com/browse/MC-197860
+ */
+@Mixin(LevelStorage.class)
+abstract class LevelStorageMixin {
+	/*
+	 * Why?
+	 * When the custom dimension is removed, DimensionGeneratorSettings.CODEC fails to fully parse the Dynamic<T>,
+	 * so the DataResult<DimensionGeneratorSettings> only return the Either#Right part of the result,
+	 * which only includes the overworld.
+	 *
+	 * This causes all dimensions to cease to exist.
+	 * To fix this issue, we will intercept the dynamic being parsed, and remove any dimension entries that are NOT in the registry.
+	 */
+
+	/**
+	 * @param dynamic the dynamic containing the properties
+	 * @param dataFixer the data fixer
+	 * @param version the data version of the properties being read
+	 * @param info the callback info
+	 * @param propertiesDynamic the dynamic containing the world gen settings
+	 * @param <T> the type in the dynamic
+	 */
+	@Inject(method = "readGeneratorProperties", at = @At(value = "INVOKE", target = "Lcom/mojang/datafixers/DataFixer;update(Lcom/mojang/datafixers/DSL$TypeReference;Lcom/mojang/serialization/Dynamic;II)Lcom/mojang/serialization/Dynamic;"), locals = LocalCapture.CAPTURE_FAILHARD)
+	private static <T> void fixDimensionsOnLoad(Dynamic<T> dynamic, DataFixer dataFixer, int version, CallbackInfoReturnable<Pair<GeneratorOptions, Lifecycle>> info, Dynamic<T> propertiesDynamic) {
+		CompoundTag properties;
+
+		// Fix the dynamic in place if the value is nbt
+		if (dynamic.getValue() instanceof CompoundTag) {
+			properties = (CompoundTag) dynamic.getValue();
+		} else {
+			// Not supported atm
+			throw new IllegalStateException("Dynamic is not backed by a Compound Tag! This is not supported at the moment.");
+		}
+
+		// Get the dimension registry
+		DataResult<Pair<Registry<DimensionType>, T>> result = RegistryLookupCodec.of(Registry.DIMENSION_TYPE_KEY).codec().decode(propertiesDynamic);
```

Comment:
Is the `codec` returning an element codec? Just a side note if this should be updated in yarn.

Reply:
This is pretty much absuing a codec to access the registry from the dynamic registry manager in registry ops from what I see.

Reply:
Lol no, this is [a mojang name](https://github.com/Mojang/DataFixerUpper/blob/e843dac1eb96af7b86b1e11165afeca7caffb7e6/src/main/java/com/mojang/serialization/MapCodec.java#L105)! I find confidence in yarn again